swap/swap.go

```diff
@@ -561,13 +591,21 @@ func (s *Swap) saveBalance(p enode.ID, balance int64) error {
 	return s.store.Put(balanceKey(p), balance)
 }
 
+func loadPaidout(s state.Store) (paidOut uint64, err error) {
+	err = s.Get(paidoutKey, &paidOut)
+	if err == state.ErrNotFound {
+		return 0, nil
+	}
+	return paidOut, err
```

Comment:
although not strictly necessary, it's customary to return the zero-value for a variable type in the case of a non-nil error being returned as the secondary element in the tuple

Reply:
The comment is outdated, but can you elaborate on what you mean here? How would you prefer the return to look?

Reply:
sure:

```golang
func loadPaidout(s state.Store) (paidOut uint64, err error) {
	err = s.Get(paidoutKey, &paidOut)
	if err != nil && err != state.ErrNotFound {
		return 0, err
	}
	return paidOut, nil
}
```
the main difference with the code (as it was when i made the comment) is that this one does not allow `paidOut` to be `0` and `err` to be different from `nil` at the same time—the previous version (theoretically) allowed a non-zero integer returned along with a non-nil error.

this isn't a big deal because:
1. it's unlikely that `paidOut` has a non-zero value if an error was produced and returned in `err`.
2. whoever called `loadPaidout` will probably check the error part of the tuple first, and in the case of a non-nil one most likely ignore the number returned as the first part of the tuple.

but it's marginally safer this way.

Reply:
Amazing, thanks